language/tools/move-cli/src/base/docgen.rs

```diff
@@ -1,48 +1,125 @@
 // Copyright (c) The Move Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::base::prove::{Prove, ProverOptions};
+use super::reroot_path;
 use clap::*;
-use move_package::BuildConfig;
-use std::path::PathBuf;
+use move_docgen::DocgenOptions;
+use move_package::{BuildConfig, ModelConfig};
+use std::{fs, path::PathBuf};
 
-/// Generating javadoc style documentation for packages using the Move Prover
+/// Generate javadoc style documentation for Move packages
 #[derive(Parser)]
 #[clap(name = "docgen")]
 pub struct Docgen {
-    /// Do not print out other than errors
-    #[clap(long = "quiet", short = 'q')]
-    pub quiet: bool,
-    /// A template for documentation generation.
+    /// The level where we start sectioning. Often markdown sections are rendered with
```

Comment:
I wonder whether it would be better to "clapify" the existing `DocgenOptions` and flatten them into the Docgen command?

Reply:
@wrwg That's a good point, that way we could avoid reduplication :+1: Will do that!

Reply:
@wrwg To me it looks like that by flattening we would lose the current "divert only when not default" model. ("_The enable/disable logic is modeled against current default settings for Docgen. This is also an indirect way to communicate the default settings to the user._"). User would need to figure out themselves what are the current defaults, and if to diver from those. This could make it a tad harder for the user: first to figure out the defaults, and then divert from those when/if needed.

Is this OK?

Reply:
Sorry for the delay. I'm not sure I understand. Aren't the current options also equipped with defaults via clap, and aren't those not displayed to the user on help? In any case if you think we need to duplicate this for usability reasons that is also fine with me.

Reply:
@wrwg To me the problem seems to be, that currently (because `move_docgen::Docgen` haven't been clap'd before) the options are mostly [booleans without default values](https://github.com/move-language/move/blob/e2bb7e9822a6c15154149599f3c44f34258b72c0/language/move-prover/move-docgen/src/docgen.rs#L43). Instead defaults are defined separately in [`move_docgen::DocgenOptions::default()`](https://github.com/move-language/move/blob/e2bb7e9822a6c15154149599f3c44f34258b72c0/language/move-prover/move-docgen/src/docgen.rs#L100).

But maybe the true long term solution would be deleting `move_docgen::DocgenOptions::default()` and use clap'd defaults instead? This would require some refactoring, though. Maybe in next iteration? What do you think?

Reply:
I think deleting `default` and using clap defaults (which is fine to do in a follow-up) makes sense. The doc functionality is quite hidden/underused right now, so I'm not too worried about folks being surprised by changes in defaults.

But the docgen functionality is super valuable, so I am glad to see this made more prominent!

Reply:
I don't understand why we introduce a total new options type here instead of refactoring the old, but fine for me to merge.